tests/test_cases/test_sim_standalone/test_sim_standalone.py

```diff
@@ -0,0 +1,194 @@
+            return -12 # pS
+
+        def get_sim_time(self):
+            return (self.time_ps >> 32, self.time_ps & 0xFFFFFFFF)
+
+    class MB():
+        def __init__(self):
+            self.p2c_ev = Event()
+            self.c2p_ev = Event()
+            self.p2c = 0
+            self.c2p = 0
+
+        def send_p2c(self):
+            self.p2c_ev.set()
+
+        @cocotb.coroutine
+        def get_c2p(self):
+            yield self.c2p_ev.wait()
+            self.c2p_ev.clear()
+            self.c2p += 1
+
+        def send_c2p(self):
+            self.c2p_ev.set()
+
+        @cocotb.coroutine
+        def get_p2c(self):
+            yield self.p2c_ev.wait()
+            self.p2c_ev.clear()
+            self.p2c += 1
+
+    def test_simple_pingpong(self):
+        """Runs a series of untimed ping-pongs"""
+
+        @cocotb.coroutine
+        def producer(mb):
+            for i in range(16):
+                mb.send_p2c()
+                yield mb.get_c2p()
+
+        @cocotb.coroutine
+        def consumer(mb):
+            for i in range(16):
+                yield mb.get_p2c()
+                mb.send_c2p()
+
+        @cocotb.test()
+        def test_main():
+            mb = TestSimStandalone.MB()
+            t_c1 = cocotb.fork(producer(mb))
+            t_c2 = cocotb.fork(consumer(mb))
+            yield [t_c1.join(), t_c2.join()]
+            self.assertEqual(mb.p2c, 16)
+            self.assertEqual(mb.c2p, 16)
+
+        sim = TestSimStandalone.SimulatorStub()
+        cocotb.simulator = sim
+        cocotb.scheduler = Scheduler()
+        cocotb.log = SimLog("cocotb")
+        cocotb.argv = []
+        cocotb.process_plusargs()
+        cocotb.log.setLevel(logging.INFO)
+
+        cocotb.scheduler.add_test(test_main())
+
+        # This will run the test-completed callback
+        sim.run()
+
+
+    def test_timed_pingpong(self):
+        """Runs a series of timed ping-pongs"""
+
+        @cocotb.coroutine
+        def producer(mb):
+            for i in range(16):
+                mb.send_p2c()
+                yield mb.get_c2p()
+                yield Timer(10)
+
+        @cocotb.coroutine
+        def consumer(mb):
+            for i in range(16):
+                yield mb.get_p2c()
+                mb.send_c2p()
+                yield Timer(20)
+
+        @cocotb.test()
+        def test_main(mb):
+            t_c1 = cocotb.fork(producer(mb))
+            t_c2 = cocotb.fork(consumer(mb))
+            yield [t_c1.join(), t_c2.join()]
+            self.assertEqual(mb.p2c, 16)
+            self.assertEqual(mb.c2p, 16)
+
+        mb = TestSimStandalone.MB()
+        sim = TestSimStandalone.SimulatorStub()
+
+        cocotb.simulator = sim
+        cocotb.scheduler = Scheduler()
+        cocotb.fork = cocotb.scheduler.add
+        cocotb.log = SimLog("cocotb")
+        cocotb.argv = []
+        cocotb.process_plusargs()
+        cocotb.log.setLevel(logging.INFO)
+
+        cocotb.scheduler.add_test(test_main(mb))
```

Comment:
I would assume that these set of lines are also hidden from the end user and he only needs to call something like.
if standalone_sim:
  set_simulator(STANDALONE)
else:
  set_simulator(DUT) # Default case?